pkg/kubectl/cmd/set/set_resources.go

```diff
@@ -141,18 +141,23 @@ func (o *ResourcesOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args
 		return err
 	}
 
-	builder := resource.NewBuilder(o.Mapper, f.CategoryExpander(), o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+	clientMapper := resource.ClientMapperFunc(f.ClientForMapping)
+
+	var builder *resource.Builder
+	if o.Local {
+		builder = resource.NewBuilder(o.Mapper, f.CategoryExpander(), o.Typer, resource.DisabledClientForMapping{ClientMapper: clientMapper}, f.Decoder(true))
```

Comment:
We're chasing this problem everywhere.  I've seen at least one other pull related to this in a different set command.   Can you find the existing factory methods for creating builders and make one flavor for "normal" operation and another for local operation and sweep the command?